processing/src/test/java/io/druid/query/aggregation/AggregationTestHelper.java

```diff
@@ -237,19 +238,57 @@ public void createIndex(
       int maxRowCount
   ) throws Exception
   {
-    try(IncrementalIndex index = new OnheapIncrementalIndex(minTimestamp, gran, metrics, deserializeComplexMetrics, maxRowCount)) {
-      while (rows.hasNext()) {
+    IncrementalIndex index = null;
+    List<File> toMerge = new ArrayList<>();
 
+    try {
+      index = new OnheapIncrementalIndex(minTimestamp, gran, metrics, deserializeComplexMetrics, maxRowCount);
+      while (rows.hasNext()) {
         Object row = rows.next();
-        if (row instanceof String && parser instanceof StringInputRowParser) {
-          //Note: this is required because StringInputRowParser is InputRowParser<ByteBuffer> as opposed to
-          //InputRowsParser<String>
-          index.add(((StringInputRowParser) parser).parse((String) row));
-        } else {
-          index.add(parser.parse(row));
+        try {
+          if (row instanceof String && parser instanceof StringInputRowParser) {
+            //Note: this is required because StringInputRowParser is InputRowParser<ByteBuffer> as opposed to
+            //InputRowsParser<String>
+            index.add(((StringInputRowParser) parser).parse((String) row));
+          } else {
+            index.add(parser.parse(row));
+          }
         }
+        catch (IndexSizeExceededException ex) {
+          File tmp = Files.createTempDir();
```

Comment:
The indexing tests are notoriously dirty about leaving files behind. Can this be reorchestrated to use a controlled TemporaryFolder @Rule, either here or from the caller?